src/nvim/os/fs.c

```diff
@@ -391,6 +391,37 @@ int os_open(const char* path, int flags, int mode)
   return r;
 }
 
+/// Set a file to close on exec.
+//
+// @return -1 if failed to set, 0 otherwise.
+int os_set_cloexec(const int fd)
+{
+#ifdef HAVE_FD_CLOEXEC
+  int e;
+  int fdflags = fcntl(fd, F_GETFD);
+  if (fdflags < 0) {
+    e = errno;
+    ELOG("Failed to get flags on descriptor %d: %s", fd, strerror(errno));
```

Comment:
Macro `ELOG()` only replaces this line with  `do_log()` and  adds some more parameters.
So this should work with `errno` instead of `e`, because no other function is called before `strerror()`.